public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx

```diff
@@ -147,3 +145,52 @@ export function buildPanelEditScene(panel: VizPanel): PanelEditor {
     }),
   });
 }
+
+// This function is used to conditionally add the data pane to the panel editor,
+// depending on the type of a panel being edited.
+function conditionalDataPaneBehavior(scene: SplitLayout) {
+  const dashboard = getDashboardSceneFor(scene);
+
+  const editor = dashboard.state.editPanel;
+
+  if (!editor) {
+    return;
+  }
+
+  const panelManager = editor.state.panelRef.resolve();
+  const panel = panelManager.state.panel;
+
+  if (!config.panels[panel.state.pluginId].skipDataQuery) {
+    scene.setState({
+      secondary: new SceneFlexItem({
+        body: new PanelDataPane(panelManager),
+      }),
```

Comment:
nipick: maybe extract this part to a variable so it can be reused on line 185 (in case we update this and miss that it's defined twice